lib/src/winrt_com_interop_helpers.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Helper wrapper classes for WinRT COM interop interfaces.
+// See https://learn.microsoft.com/en-us/windows/apps/develop/ui-input/display-ui-objects
+
+import 'com/iinitializewithwindow.dart';
+import 'com/iinspectable.dart';
+import 'exceptions.dart';
+import 'macros.dart';
+import 'win32/user32.g.dart';
+
+/// Exposes a method through which a client can provide an owner window to a
+/// Windows Runtime (WinRT) object used in a desktop application.
+///
+/// {@category Class}
+/// {@category winrt}
+class InitializeWithWindow {
+  /// Specifies an owner window to be used by a Windows Runtime object that is
+  /// used in a desktop app.
+  ///
+  /// [target] must be a WinRT object that implements the
+  /// `IInitializeWithWindow` interface (e.g. `FileOpenPicker`).
+  ///
+  /// [hwnd] represents the handle of the window to be used as the owner window.
```

Comment:
I'm curious if there are only specific types of `HWND` that are valid. I used `GetActiveWindow()` (which I would have thought would be the natural choice), and got an error: `Error 0x80070578: Invalid window handle.`, which suggests that there's some specific criteria beyond being a valid `HWND`. I wasn't able to find documentation that answered this beyond [this article](https://learn.microsoft.com/en-us/windows/apps/develop/ui-input/display-ui-objects).

Reply:
> I'm curious if there are only specific types of HWND that are valid. I used GetActiveWindow() (which I would have thought would be the natural choice), and got an error: Error 0x80070578: Invalid window handle., which suggests that there's some specific criteria beyond being a valid HWND. I wasn't able to find documentation that answered this beyond [this article](https://learn.microsoft.com/en-us/windows/apps/develop/ui-input/display-ui-objects).

I don't know. I found `GetConsoleWindow()` and `GetShellWindow()` by trial and error.

I just noticed that `GetForegroundWindow()` works for both console and Flutter apps (_yay!_). I wonder, should I make the necessary changes so that we use `GetForegroundWindow()` everywhere?

Reply:
`GetForegroundWindow()` is good, but it's not perfect -- for example, if there's another window from another process that has gained focus, it will become the parent for the file picker. That's why I'm a bit bemused that `GetActiveWindow()` isn't working here, since that means "the current window from the current process". (More here: https://devblogs.microsoft.com/oldnewthing/20081006-00/?p=20643)

Reply:
Ah, I see. It was too good to be true anyway :).

Reply:
This seems to explain it:
https://devblogs.microsoft.com/oldnewthing/20190412-00/?p=102413

(TL;DR Not every window has a `CoreWindow` associated with it. We ought to figure out how to do `IInitializeWithWindow` from Dart).